### PR TITLE
feat: added diff info collection

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

collected info will be registered to `arstotzka` and will look like this:
```json
{
  "info": {
    "data": {
      "bbox": [
        1.5201119,
        42.5079796,
        1.5338618,
        42.5408647
      ],
      "count": {
        "ways": 7,
        "nodes": 10,
        "relations": 1,
        "changesets": 0
      },
      "crc32": "a34ab64d",
      "maxid": {
        "ways": 1199455446,
        "nodes": 11125044850,
        "relations": 3004189,
        "changesets": 0
      },
      "minid": {
        "ways": 144382952,
        "nodes": 51399278,
        "relations": 3004189,
        "changesets": 0
      },
      "buffers": {
        "size": 2840,
        "count": 1,
        "capacity": 1048576
      },
      "metadata": {
        "all_objects": {
          "uid": false,
          "user": false,
          "version": true,
          "changeset": false,
          "timestamp": true
        },
        "some_objects": {
          "uid": false,
          "user": false,
          "version": true,
          "changeset": false,
          "timestamp": true
        }
      },
      "timestamp": {
        "last": "2023-08-18T18:44:22Z",
        "first": "2023-08-18T11:49:57Z"
      },
      "objects_ordered": true,
      "multiple_versions": false
    },
    "file": {
      "name": "/tmp/005/055/001.osc.gz",
      "size": 1246,
      "format": "XML",
      "compression": "gzip"
    },
    "header": {
      "boxes": [],
      "option": {
        "version": "0.6",
        "generator": "osmium/1.14.0"
      },
      "with_history": true
    }
  }
}
```

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -27,6 +29,7 @@ export const ExitCodes = {
   SUCCESS: 0,
   GENERAL_ERROR: 1,
   OSMDBT_ERROR: 100,
+  OSMIUM_ERROR: 101,
```
Comment (syncush) [inline src/constants.ts:32]: update in README ?

**Review by syncush — CHANGES_REQUESTED** [review]

**Comment by syncush** [comment]
CI fails on security , please check it

File: src/constants.ts (hunk 1 above)
Comment (melancholiai) [reply src/constants.ts:32]: done.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -38,6 +38,7 @@ jobs:
     - uses: actions/checkout@master
     - name: Run Snyk to check for vulnerabilities
       uses: snyk/actions/node@master
+      continue-on-error: true
```
Comment (syncush) [inline .github/workflows/pull_request.yml:41]: I prefer to the test to fails and see the report then continue on error, even if it exists in other repos.

**Review by syncush — CHANGES_REQUESTED** [review]

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (melancholiai) [reply .github/workflows/pull_request.yml:41]: this is a known [issue](https://github.com/actions/runner/issues/2347) which sadly github-workflow does not support. 
Notice that the report is shown but the step is marked with a green tick.

**Pull request merged**